packages/browser-wallet/src/popup/pages/Account/ConnectedBox.tsx

```diff
@@ -0,0 +1,52 @@
+import { useAtomValue } from 'jotai';
+import React, { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import { storedConnectedSitesAtom } from '@popup/store/account';
+import { getCurrentOpenTabUrl } from '@popup/shared/utils/tabs';
+import clsx from 'clsx';
+
+type Props = {
+    accountAddress?: string;
+    getUrl?: () => Promise<string>;
+    link?: string;
+    onNavigate?: () => void;
+};
+
+/**
+ * N.B. It is assumed that onNavigate is only given when the box has a link. If link is not provided, then onNavigate is ignored.
```

Comment:
It would be preferable to make a type that couples `{ link, onNavigate }` together as a pair, so that you don't need this disclaimer.